test/Web/Rollbar/Test.hs

```diff
@@ -0,0 +1,118 @@
+{-# LANGUAGE DeriveGeneric #-}
+{-# LANGUAGE OverloadedStrings #-}
+{-# LANGUAGE TemplateHaskell #-}
+
+module Web.Rollbar.Test where
+
+import Web.Rollbar
+    ( AccessToken(..)
+    , CodeVersion(..)
+    , Environment(..)
+    , Event(..)
+    , EventLevel(..)
+    , HasRollbarCfg(..)
+    , Host(..)
+    , RollbarCfg(..)
+    )
+import qualified Web.Rollbar.Internal as I
+
+import Control.Lens.TH (makeLenses)
+import Control.Monad.Reader (runReaderT)
+import Data.Aeson (ToJSON(..), defaultOptions, genericToEncoding)
+import Data.Aeson.Encode.Pretty (Config(..), Indent(..), defConfig, encodePretty')
+import Data.ByteString.Lazy (ByteString)
+import Data.Monoid ((<>))
+import Data.Text (Text)
+import qualified Data.Text.IO as TIO
+import GHC.Generics (Generic)
+import Test.Tasty (TestTree, testGroup)
+import Test.Tasty.Golden (goldenVsFile, goldenVsString, writeBinaryFile)
+
+newtype Context = Context
+    { _ctxRollbarCfg :: RollbarCfg
+    }
+
+makeLenses ''Context
+
+instance HasRollbarCfg Context where
+    rollbarCfg = ctxRollbarCfg
```

Comment:
I have little experience with Template Haskell and lenses and was wondering if this is a reasonable setup for testing? Please let me know if there is a better way to get the `MonadReader` for `encodeEvent` set up. Thanks 🙇

Reply:
Personally, I would have implemented it manually, just because I try not to use TH if I can, I find it timey-wimey. 
But this way is also good, arguably even clearer.

Reply:
Thanks! Do we have an example of writing lenses / `Has*` typeclasses manually? Or did you mean something else?

Reply:
[Here](https://github.com/collegevine/vines/blob/master/cv-base/src/CollegeVine/Context/Instances.hs#L39) are a bunch of them, but that's a bit confusing, because they're based on another, more general lens. A lot of unnecessary boilerplate, should be generalized.

But basically, you just apply the `lens` constructor to a getter and a setter, e.g.:

```
instance HasRollbarCfg Context where
    rollbarCfg = lens get' set' where
        get' = _ctxRollbarCfg
        set' ctx cfg = ctx { _ctxRollbarCfg = cfg }
```

Reply:
Perfect, thanks!